server/src/test/java/org/opensearch/action/admin/cluster/snapshots/get/GetSnapshotsActionTests.java

```diff
@@ -0,0 +1,245 @@
+        );
+
+        return typesRegistry;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
+        threadPool = null;
+        clusterService.close();
+        super.tearDown();
+    }
+
+    public void testRepoMissing() {
+        GetSnapshotsRequest repoMissingRequest = new GetSnapshotsRequest();
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().snapshots(snapshotsName1);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().ignoreUnavailable(true);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().verbose(true);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+
+        repoMissingRequest = new GetSnapshotsRequest().verbose(false);
+
+        getSnapshotsAction.execute(null, repoMissingRequest, ActionListener.wrap(repoMissingResponse -> {
+            assertThrows(ActionRequestValidationException.class, () -> { repoMissingResponse.getSnapshots(); });
+        }, exception -> { assertTrue(exception.getMessage().contains("repository is missing")); }));
+    }
+
+    public void testCurrentRepoSnapshot() {
+        final Task task = createTask();
+        RepositoryMetadata repositoryMetadata = new RepositoryMetadata(repoName, repoType, Settings.EMPTY);
+        Repository repository = repositoriesService.createRepository(repositoryMetadata, repositoriesService.typesRegistry());
+        ;
+        Map<String, Repository> repositoryMap = new HashMap<>();
+        repositoryMap.put(repoName, repository);
+        repositoriesService = repositoriesService.repositories(repositoryMap);
+        assertEquals(repositoryMap, repositoriesService.repositories());
```

Comment:
no sure what are you trying to do here, we dont need to test RepositoriesService here.

Reply:
Ack. This was checked to complete the codeCov check, as the similar method was exposed on the RepositoriesService call.

Reply:
This has been revoked.